### PR TITLE
ICS27 Spec Improvements

### PR DESCRIPTION
**Pull request by AdityaSripal**
Addresses remaining issues from #618 and #616 reviews

### CODE REVIEW

**Review by seantking — APPROVED** [review]

**Comment by seantking** [comment]
@AdityaSripal Many thanks :handshake: 

@adizere @cwgoes A follow-up to discuss some of your concerns & address prior comments.

File: spec/app/ics-027-interchain-accounts/README.md [hunk 1]
```diff
@@ -130,14 +155,9 @@ function (k Keeper) DeleteActiveChannelID(portID string) {
...
+1. The controller chain binds a new IBC port with the controller portID for a given *source/counterparty connection-ids* and *interchain account owner address*.
 
-The IBC port identifier will look like this:
-```
-ics27-1.{connection-id}.{counterparty-connection-id}.{owner-account-address}
-```
-
-This port will be used to create channels between the controller & host chain for a specific owner/interchain account pair. Only the account with `{owner-account-address}` matching the bound port will be authorized to send IBC packets over channels created with `ics27-1.{connection-id}.{counterparty-connection-id}.{owner-account-address}`. It is up to each controller chain to enforce this port registration and access on the controller side. 
+This port will be used to create channels between the controller & host chain for a specific owner/interchain account pair. Only the account with `{owner-account-address}` matching the bound port will be authorized to send IBC packets over channels created with the controller portID. It is up to each controller chain to enforce this port registration and access on the controller side.
```
Comment (cezarad) [inline spec/app/ics-027-interchain-accounts/README.md:160]: When a channel is reopen it uses the same port ids  (as the previous one) and a new channel-id  ?

File: spec/app/ics-027-interchain-accounts/README.md [hunk 2]
```diff
@@ -420,10 +449,24 @@ function onTimeoutPacket(packet: Packet) {
...
+
 ## Example Implementation
 
 Repository for Cosmos-SDK implementation of ICS-27: https://github.com/cosmos/ibc-go
 
+## Future Improvements
+
+A future version of interchain accounts may be greatly simplified by the introduction of an IBC channel type that is ORDERED but does not close the channel on timeouts, and instead proceeds to accept and receive the next packet. If such a channel type is made available by core IBC, Interchain accounts could require the use of this channel type and remove all logic and state pertaining to "active channels". The controller port identifier format can also be simplified to remove any reference to the underlying connection identifiers
+
+The "active channel" setting and unsetting is currently necessary to allow interchain account owners to create a new channel in case the current active channel closes during channel timeout. The connection identifiers are part of the portID to ensure that any new channel that gets opened are established on top of the original connection. All of this logic becomes unnecessary once the channel is ordered **and** unclosable, which can only be achieved by the introduction of a new channel type to core IBC.
```
Comment (cezarad) [inline spec/app/ics-027-interchain-accounts/README.md:468]: "any new channel that gets opened IS established"

File: spec/app/ics-027-interchain-accounts/README.md [hunk 3]
```diff
@@ -46,61 +46,88 @@ A chain can utilize one or both parts of the interchain accounts protocol (*cont
...
 
+#### **InitChannel**
 
-// Opens a new channel on a particular port given a connection
-// This is a helper function to open a new channel 
-// This is a safety function in case of a channel closing and the controller chain needs to regain access to an interchain account on the host chain 
+Opens a new channel on a particular port given a connection.
+This is a helper function to open a new channel.
+This is a safety function in case of a channel closing and the controller chain needs to regain access to an interchain account on the host chain.
+
```
[the remark this answers is not in the document]
Comment (AdityaSripal) [reply spec/app/ics-027-interchain-accounts/README.md:82]: It is not for reopening a channel. A channel cannot be reopened.

We are "reactivating" the interchain account by creating a **new** channel that will be associated with it.

File: spec/app/ics-027-interchain-accounts/README.md (hunk 1 above)
Comment (AdityaSripal) [reply spec/app/ics-027-interchain-accounts/README.md:160]: Correct. Though this is a completely new channel

File: spec/app/ics-027-interchain-accounts/README.md [hunk 4]
```diff
@@ -420,10 +449,24 @@ function onTimeoutPacket(packet: Packet) {
...
+Host Port Identifier: `interchain-account`
+
 ## Example Implementation
 
 Repository for Cosmos-SDK implementation of ICS-27: https://github.com/cosmos/ibc-go
 
+## Future Improvements
+
+A future version of interchain accounts may be greatly simplified by the introduction of an IBC channel type that is ORDERED but does not close the channel on timeouts, and instead proceeds to accept and receive the next packet. If such a channel type is made available by core IBC, Interchain accounts could require the use of this channel type and remove all logic and state pertaining to "active channels". The controller port identifier format can also be simplified to remove any reference to the underlying connection identifiers
```
Comment (cwgoes) [inline spec/app/ics-027-interchain-accounts/README.md:466]: Thank you, this explanation is very helpful.

**Review by cwgoes — APPROVED** [review]
ACK, thank you!

**Pull request merged**